mlflow/types/utils.py

```diff
@@ -17,45 +17,60 @@ def __init__(self, msg):
         )
 
 
+def _get_tensor_shape(data: np.ndarray) -> tuple:
+    """
+    Infer the shape of the inputted data.
+
+    This method creates the shape of the tensor to store in the TensorSpec. Always returns a shape
+    including a variable dimension. Assumes the variable dimension is the first dimension.
+
+    NB: For 1D tensors, the output is (-1, shape).
+
+    :param data: Dataset to infer from.
+    :return: tuple : Shape of the inputted data (including a variable dimension)
+    """
+    if not isinstance(data, np.ndarray):
+        raise TypeError("Expected numpy.ndarray, got '{}'.".format(type(data)))
+    if len(data.shape) == 1:
```

Comment:
hmm I don't think we need this branch. If the input is a 1-d np array we should just return [-1]. The else branch takes care of that.